src/v/raft/consensus.cc

```diff
@@ -908,7 +909,30 @@ ss::future<> consensus::start() {
     vlog(_ctxlog.info, "Starting");
     return _op_lock.with([this] {
         read_voted_for();
-        const bool initial_state = is_initial_state();
+
+        /*
+         * temporary workaround:
+         *
+         * if the group's ntp matches the pattern, then do not load the initial
+         * configuration snapshto from the keyvalue store. more info here:
+         *
+         * https://github.com/vectorizedio/redpanda/issues/1870
+         */
+        const auto& ntp = _log.config().ntp();
+        const auto normalized_ntp = fmt::format(
+          "{}.{}.{}", ntp.ns(), ntp.tp.topic(), ntp.tp.partition());
+        const auto& patterns = config::shard_local_cfg()
+                                 .full_raft_configuration_recovery_pattern();
+        auto initial_state = std::any_of(
```

Comment:
that's cool, I like that we can turn the workaround just for a fraction of the partitions